Business understanding
The data we`ve been provided with focuses on the feedbacks of passengers in different airlines. The stakeholders here are; Virgin America, American, Turkish airlines, The flight customers, The airports where these airlines operate and who collected the data in form of tweets.
The overall conclusion from the feedbacks is that the customers are not satisfied with how the flights operate. We have more negative feedbacks than positive and neutral combined. We are interested to know why are the flights delayed. How are passengers treated right from check in to during flights. What needs to be changed to be done to improve the present conditions and services so as to get more positive feedback from the customers.
Our success metric in this case is the ratio of positive feedback to that of negative and neutral combined. The positive reviews should outbalance the other two for it to be a success. In our project we assume that all the six airlines have the same operating business models and the conclusions that we are going to draw from this project are going to cut across all the airlines. The limitation we have here is that we lack data from the airline companies. For instance, maybe the flight was delayed due to valid reasons like hostile weather conditions or security issues.



In [1]:
import csv
import pandas as pd
df=pd.read_csv('Tweets.csv')
df

tweet_id airline_sentiment  airline_sentiment_confidence  \
0      570306133677760513           neutral                        1.0000   
1      570301130888122368          positive                        0.3486   
2      570301083672813571           neutral                        0.6837   
3      570301031407624196          negative                        1.0000   
4      570300817074462722          negative                        1.0000   
...                   ...               ...                           ...   
14635  569587686496825344          positive                        0.3487   
14636  569587371693355008          negative                        1.0000   
14637  569587242672398336           neutral                        1.0000   
14638  569587188687634433          negative                        1.0000   
14639  569587140490866689           neutral                        0.6771   

               negativereason  negativereason_confidence         airline  \
0                         NaN                        NaN  Virgin America   
1                         NaN                     0.0000  Virgin America   
2                         NaN                        NaN  Virgin America   
3                  Bad Flight                     0.7033  Virgin America   
4                  Can't Tell                     1.0000  Virgin America   
...                       ...                        ...             ...   
14635                     NaN                     0.0000        American   
14636  Customer Service Issue                     1.0000        American   
14637                     NaN                        NaN        American   
14638  Customer Service Issue                     0.6659        American   
14639                     NaN                     0.0000        American   

      airline_sentiment_gold             name negativereason_gold  \
0                        NaN          cairdin                 NaN   
1                        NaN         jnardino                 NaN   
2                        NaN       yvonnalynn                 NaN   
3                        NaN         jnardino                 NaN   
4                        NaN         jnardino                 NaN   
...                      ...              ...                 ...   
14635                    NaN  KristenReenders                 NaN   
14636                    NaN         itsropes                 NaN   
14637                    NaN         sanyabun                 NaN   
14638                    NaN       SraJackson                 NaN   
14639                    NaN        daviddtwu                 NaN   

       retweet_count                                               text  \
0                  0                @VirginAmerica What @dhepburn said.   
1                  0  @VirginAmerica plus you've added commercials t...   
2                  0  @VirginAmerica I didn't today... Must mean I n...   
3                  0  @VirginAmerica it's really aggressive to blast...   
4                  0  @VirginAmerica and it's a really big bad thing...   
...              ...                                                ...   
14635              0  @AmericanAir thank you we got on a different f...   
14636              0  @AmericanAir leaving over 20 minutes Late Flig...   
14637              0  @AmericanAir Please bring American Airlines to...   
14638              0  @AmericanAir you have my money, you change my ...   
14639              0  @AmericanAir we have 8 ppl so we need 2 know h...   

      tweet_coord              tweet_created tweet_location  \
0             NaN  2015-02-24 11:35:52 -0800            NaN   
1             NaN  2015-02-24 11:15:59 -0800            NaN   
2             NaN  2015-02-24 11:15:48 -0800      Lets Play   
3             NaN  2015-02-24 11:15:36 -0800            NaN   
4             NaN  2015-02-24 11:14:45 -0800            NaN   
...           ...                        ...            ...   
14635   

In [2]:
df.info

<bound method DataFrame.info of                  tweet_id airline_sentiment  airline_sentiment_confidence  \
0      570306133677760513           neutral                        1.0000   
1      570301130888122368          positive                        0.3486   
2      570301083672813571           neutral                        0.6837   
3      570301031407624196          negative                        1.0000   
4      570300817074462722          negative                        1.0000   
...                   ...               ...                           ...   
14635  569587686496825344          positive                        0.3487   
14636  569587371693355008          negative                        1.0000   
14637  569587242672398336           neutral                        1.0000   
14638  569587188687634433          negative                        1.0000   
14639  569587140490866689           neutral                        0.6771   

               negativereason  negativereas

In [4]:
df.shape

(14640, 15)

Data understanding
We are dealing with data already collected by someone else and saved to a file in form of csv. The source of this data was twitter on the feedback section from passengers. The data has 14640 rows and 15 columns. The columns contain sections where we have the name of the airline, the actual feedback, name of the passenger, tweet ID, time and location of the tweet. The confidence on the airline based on the sentiments provided have been put in a scale of 0-1 with 1 representing the highest confidence level.


In [7]:
df.duplicated().sum()

36

In [9]:
print(df[df.duplicated()])

                 tweet_id airline_sentiment  airline_sentiment_confidence  \
12040  570308309682675712          negative                           1.0   
12059  570305051819941889           neutral                           1.0   
12063  570303383782989824           neutral                           1.0   
12066  570301929580048385           neutral                           1.0   
12068  570301395141836801          negative                           1.0   
12069  570300915418320897          negative                           1.0   
12074  570299252141903873          positive                           1.0   
12075  570298770136674304          negative                           1.0   
12078  570298644475346945          negative                           1.0   
12079  570298371140939776          negative                           1.0   
12086  570296996445204480          negative                           1.0   
12093  570295576446808065          negative                           1.0   

In [10]:
df.isnull()

tweet_id  airline_sentiment  airline_sentiment_confidence  \
0         False              False                         False   
1         False              False                         False   
2         False              False                         False   
3         False              False                         False   
4         False              False                         False   
...         ...                ...                           ...   
14635     False              False                         False   
14636     False              False                         False   
14637     False              False                         False   
14638     False              False                         False   
14639     False              False                         False   

       negativereason  negativereason_confidence  airline  \
0                True                       True    False   
1                True                      False    False   
2                True                       True    False   
3               False                      False    False   
4               False                      False    False   
...               ...                        ...      ...   
14635            True                      False    False   
14636           False                      False    False   
14637            True                       True    False   
14638           False                      False    False   
14639            True                      False    False   

       airline_sentiment_gold   name  negativereason_gold  retweet_count  \
0                        True  False                 True          False   
1                        True  False                 True          False   
2                        True  False                 True          False   
3                        True  False                 True          False   
4                        True  False                 True          False   
...                       ...    ...                  ...            ...   
14635                    True  False                 True          False   
14636                    True  False                 True          False   
14637                    True  False                 True          False   
14638                    True  False                 True          False   
14639                    True  False                 True          False   

        text  tweet_coord  tweet_created  tweet_location  user_timezone  
0      False         True          False            True          False  
1      False         True          False            True          False  
2      False         True          False           False          False  
3      False         True          False            True          False  
4      False         True          False            True          False  
...      ...          ...            ...             ...            ...  
14635  False         True          False            True           True  
14636  False         True          False           False           True  
14637  False         True          False           False           True  
14638  False         True          False           False          False  
14639  False         True          False           False           True  

[14640 rows x 15 columns]

In [11]:
df.isnull().sum()

tweet_id                            0
airline_sentiment                   0
airline_sentiment_confidence        0
negativereason                   5462
negativereason_confidence        4118
airline                             0
airline_sentiment_gold          14600
name                                0
negativereason_gold             14608
retweet_count                       0
text                                0
tweet_coord                     13621
tweet_created                       0
tweet_location                   4733
user_timezone                    4820
dtype: int64

Data cleaning
This section we are going deeper to our data to pick what we need for this study and leave out that which is not important to us. There are columns like the name of the passenger, the time of the tweet and location of the tweet that we are going to filter out. Our data has lots of missing values and we are going to handle this using the data.fillna() methods. We have several duplicates and outliers in our data. The format of our data is good and we will retain that.
